R/arrange.R

```diff
@@ -17,14 +17,37 @@ arrange.duckplyr_df <- function(.data, ..., .by_group = FALSE, .locale = NULL) {
         return(.data)
       }
 
+      # Extract the sort order of arguments
+      # i.e. check if the argument is wrapped in desc()
+      ascending <- sapply(dots,function(dot) { 
+        expr <- get_expr(dot)
+        if (typeof(expr) != "language") return(TRUE)
+        return(expr[[1]] != "desc")
+      })
```

Comment:
- `map_lgl()`
- `is.call()` perhaps?
- Avoid `return()` as last statement
- Extract this and the next map into a function that returns a data frame (or a list with two parallel vectors) and uses ... a `for` loop?

Reply:
Thanks! I refactored the extraction of sort order and manipulation of dots into a separate function.

Also the parameter `ascending` in `rel_order()` now has NULL as a default argument (the tests for previous commit failed due to not having a default argument)